keps/sig-storage/2605-option-to-preserve-default-file-mode-bit-in-atomicwriter-volumes/README.md

```diff
@@ -0,0 +1,170 @@
+have the ability to access a shared\common secret or config volume.
+
+As a platform, Kubernetes should evolve to allow the sharing of AtomicWriter volumes (secret, config etc.)
+across containers running under different UIDs as a first party scenario.
+
+With this feature, we try to provide a backwards compatible way for the users to
+share a AtomicWriter volumes across containers running under different UIDs.
+
+We intend to provide a way for the users to opt-in for the new behavior and expect not to break
+any existing applications\configurations.
+
+## Proposal
+
+Kubernetes should implement a new boolean flag (PreserveDefaultMode) in the SecretVolumeSource and
+other AtomicWriter volumes.
+
+The flag should be optional and default to false in code. This flag should be honored at the time of volume
+set-up and call to SetVolumeOwnership should be skipped in case PreserveDefaultMode=true.
+
+### Preserve DefaultMode Flag
+
+A new PreserveDefaultMode boolean will be implemented in SecretVolumeSource and other AtomicWriter volumes.
+
+```go
+// SecretVolumeSource adapts a Secret into a volume.
+//
+// The contents of the target Secret's Data field will be presented in a volume
+// as files using the keys in the Data field as the file names.
+// Secret volumes support ownership management and SELinux relabeling.
+type SecretVolumeSource struct {
+    // Name of the secret in the pod's namespace to use.
+    // +optional
+    SecretName string
+    // If unspecified, each key-value pair in the Data field of the referenced
+    // Secret will be projected into the volume as a file whose name is the
+    // key and content is the value. If specified, the listed keys will be
+    // projected into the specified paths, and unlisted keys will not be
+    // present. If a key is specified which is not present in the Secret,
+    // the volume setup will error unless it is marked optional. Paths must be
+    // relative and may not contain the '..' path or start with '..'.
+    // +optional
+    Items []KeyToPath
+    // Mode bits to use on created files by default. Must be a value between
+    // 0 and 0777.
+    // Directories within the path are not affected by this setting.
+    // This might be in conflict with other options that affect the file
+    // mode, like fsGroup, and the result can be other mode bits set.
+    // If PreserveDefaultMode is set to true then file mode bit is not
+    // modified after the file creation, due to any other options set
+    // such as fsGroup etc
+    // +optional
+    DefaultMode *int32
+    // Specify whether the Secret or its key must be defined
+    // +optional
+    Optional *bool
+    // Specify whether we want to preserve the DefaultMode set on the files
+    // at the time of file creation.
+    // If PreserveDefaultMode is set to true the file mode is not modified based
+    // on fsGroup after the file is created with the defaultMode.
+    // +optional
+    PreserveDefaultMode *bool
```

Comment:
We should specify the default behaviour (i.e when field is not set for completeness sake)

Reply:
updated. Thanks!